logging/logger.go

```diff
@@ -31,13 +36,32 @@ type Logger interface {
 
 	// Log formatted messages with level DEBUG
 	Debugf(string, ...interface{})
+
+	WithFields(map[string]interface{}) Logger
 }
 
-func (dl *DefaultLog) Error(a ...interface{})            { logrus.Error(a...) }
-func (dl *DefaultLog) Errorf(f string, a ...interface{}) { logrus.Errorf(f, a...) }
-func (dl *DefaultLog) Warn(a ...interface{})             { logrus.Warn(a...) }
-func (dl *DefaultLog) Warnf(f string, a ...interface{})  { logrus.Warnf(f, a...) }
-func (dl *DefaultLog) Info(a ...interface{})             { logrus.Info(a...) }
-func (dl *DefaultLog) Infof(f string, a ...interface{})  { logrus.Infof(f, a...) }
-func (dl *DefaultLog) Debug(a ...interface{})            { logrus.Debug(a...) }
-func (dl *DefaultLog) Debugf(f string, a ...interface{}) { logrus.Debugf(f, a...) }
+func (dl *DefaultLog) Error(a ...interface{}) { dl.logger.WithFields(dl.fields).Error(a...) }
+func (dl *DefaultLog) Errorf(f string, a ...interface{}) {
+	dl.logger.WithFields(dl.fields).Errorf(f, a...)
+}
+func (dl *DefaultLog) Warn(a ...interface{}) { dl.logger.WithFields(dl.fields).Warn(a...) }
+func (dl *DefaultLog) Warnf(f string, a ...interface{}) {
+	dl.logger.WithFields(dl.fields).Warnf(f, a...)
+}
+func (dl *DefaultLog) Info(a ...interface{}) { dl.logger.WithFields(dl.fields).Info(a...) }
+func (dl *DefaultLog) Infof(f string, a ...interface{}) {
+	dl.logger.WithFields(dl.fields).Infof(f, a...)
+}
+func (dl *DefaultLog) Debug(a ...interface{}) { dl.logger.WithFields(dl.fields).Debug(a...) }
+func (dl *DefaultLog) Debugf(f string, a ...interface{}) {
+	dl.logger.WithFields(dl.fields).Debugf(f, a...)
+}
+func (dl *DefaultLog) WithFields(fields map[string]interface{}) Logger {
+	for k, v := range fields {
+		dl.fields[k] = v
+	}
+	return dl
+}
+func New() *DefaultLog {
+	return &DefaultLog{logger: *logrus.New()}
+}
```

Comment:
I think it should be something like:
```go
type DefaultLog struct {
	logger logrus.Logger
}

func (dl *DefaultLog) Error(a ...interface{}) { dl.logger.Error(a...) }
// other logger methods omitted for brevity

func New() Logger {
	return &DefaultLog{logger: logrus.StandardLogger()}
}

func (dl *DefaultLog) WithFields(fields map[string]interface{}) Logger {
	return &DefaultLog{logger: dl.logger.WithFields(logrus.Fields(fields))}
}
```
i.e. there is no need to store `fields map[string]interface{}` and call WithFields on each logging call.